### PR TITLE
Address outstanding issues

### PR DESCRIPTION
**Pull request by cbillingham**
Finally got around to addressing our outstanding issues
Also fixed a strange ast parsing bug I encountered where assign nodes were skipping the following sibling.

### CODE REVIEW

**Comment by cbillingham** [comment]
Ha github actions dropped python2.7. I suppose we should as well. I did test locally in 2.7 for kicks and made few changes so it was compatible. Still works in 2.7 technically but we should probably just drop support I think.

**Comment by cbillingham** [comment]
@AWhetter, I'm going to merge this, but please leave notes if you get a chance to review and I can address them in a second pass

File: setup.cfg [hunk 1]
```diff
@@ -18,19 +18,18 @@ classifiers =
...
     Programming Language :: Python :: 3.11
+    Programming Language :: Python :: 3.12
+    Programming Language :: Python :: 3.13
 
 [options]
 package_dir=
     =src
 packages=find:
-python_requires = >=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, !=3.5.*
+python_requires = >=3.6
```
Comment (AWhetter) [inline setup.cfg:32]: Was this supposed to be `>=3.9`?

**Review by AWhetter — COMMENTED** [review]
It's about time we dropped Python 2.7 support :D
Looks good to me!

File: setup.cfg (hunk 1 above)
Comment (cbillingham) [reply setup.cfg:32]: Oh yeah good catch

**Pull request merged**